vim/merlin/autoload/merlin.py

```diff
@@ -365,15 +365,15 @@ def command_holes():
 def vim_complete_prepare(str):
     return re.sub(re_wspaces, " ", str).replace("'", "''").strip()
 
-def vim_complete_prepare_preserve_newlines(str):
-    return re.sub(re_spaces_around_nl, "\n", re.sub(re_spaces, " ", str)).replace("'", "''").strip()
-
+# Turns Merlin's search-by-polarity or complete-prefix entries into a Vim's completion-item list
 def vim_fillentries(entries, vimvar):
-    prep = vim_complete_prepare
-    prep_nl = vim_complete_prepare_preserve_newlines
     for prop in entries:
-        vim.command("let tmp = {'word':'%s','menu':'%s','info':'%s','kind':'%s'}" %
-                (prep(prop['name']),prep(prop['desc']),prep_nl(prop['info']),prep(prop['kind'][:1])))
+        vim.command("let tmp = " + vim_record({
```

Comment:
It's unclear to me that vim_record behave similarly as the previous  use of `prep` and `prep_nl`. Does it "preserve newlines" ? And only for the "info" field ?

Reply:
You are right, there was a problem here!
I've fixed `vim_record` to properly encode newlines and added back the `prep` and `prep_nl` functions. I'm not sure in which cases these are useful.
I've just learnt from this "info" feature and I find it awesome. I tested that completion works fine with these settings:
```
let g:merlin_completion_with_doc = 1
set completeopt+=popup
```